electricitymap/contrib/lib/models/events.py

```diff
@@ -208,6 +253,59 @@ def to_dict(self) -> Dict[str, Any]:
         pass
 
 
+class AggregatableEvent(Event):
+    """
+    An abstract class representing all types of electricity events that can be aggregated.
+    """
+
+    @staticmethod
+    def _unique_zone_key(df_view: pd.DataFrame) -> ZoneKey:
+        zone_key = df_view["zoneKey"].unique()
+        if len(zone_key) > 1:
+            raise ValueError(f"Cannot merge events from different zones: {zone_key}")
+        return zone_key[0]
+
+    @staticmethod
+    def _sources(df_view: pd.DataFrame) -> str:
+        sources = df_view["source"].unique()
+        return ", ".join(sources)
+
+    @staticmethod
+    def _unique_source_type(df_view: pd.DataFrame) -> EventSourceType:
+        source_type = df_view["sourceType"].unique()
+        if len(source_type) > 1:
+            raise ValueError(
+                f"Cannot merge events from different source types: {source_type}"
+            )
+        return source_type[0]
+
+    @staticmethod
+    def _unique_datetime(df_view: pd.DataFrame) -> datetime:
+        datetime = df_view.index.unique()
+        if len(datetime) > 1:
+            raise ValueError(
+                f"Cannot merge events from different datetimes: {datetime}"
+            )
+        return datetime[0]
```

Comment:
We should probably normalize the time zones here first otherwise we might run into issues where there is diffrent timezone files but the value is the same.

Reply:
If the timezone is not the same this will result in an error anyways which is the intended behavior.
The unique part would see different objects.